src/main/proto/dcb.proto

```diff
@@ -0,0 +1,403 @@
+syntax = "proto3";
+package io.axoniq.axonserver.grpc.event.dcb;
+option java_multiple_files = true;
+
+/* EXPERIMENTAL: Service providing Event Store RPCs supporting DCB. */
+service DcbEventStore {
+
+  /* Appends new events to the store. */
+  rpc Append (stream AppendEventsRequest) returns (AppendEventsResponse);
+
+  /* Provides a finite stream of events based on the passed SourceEventsRequest. */
+  rpc Source (SourceEventsRequest) returns (stream SourceEventsResponse);
+
+  /* Provides an infinite stream of events based on the passed StreamEventsRequest.  */
+  rpc Stream (StreamEventsRequest) returns (stream StreamEventsResponse);
+
+  /* Gets the current _head_ of the Event Store. The _head_ points to a sequence of the first event to be appended. */
+  rpc GetHead (GetHeadRequest) returns (GetHeadResponse);
+
+  /* Gets the current _tail_ of the Event Store. The _tail_ points to a sequence of the first event stored. */
+  rpc GetTail (GetTailRequest) returns (GetTailResponse);
+
+  /* Gets the sequence of the event whose timestamp approximately matches the given request. It returns the first
```

Comment:
"Approximately match" is very vague. I suggest:

Returns the lowest sequence of an event with a timestamp equal to or higher than the given timestamp. The HEAD is returned if no events exist with a timestamp equal to or higher than the given timestamp.

The NB part is good. Best to warn people that the timestamps are application-provided and that Axon Server has no control over them. Note that "monotonically" is not relevant here. Monotonically means that the delta between each event's timestamp is equal. I would replace that with "strictly".

Reply:
I agree with the first part. 

```
Formal Definition of Monotonically Increasing Function:
A function f(x) is monotonically increasing if for all x1 and x2 in its domain where x1 ≤ x2, then f(x1) ≤ f(x2). 
```

The values must be greater than or equal to. It says nothing about deltas.